src/MainScene.gd

```diff
@@ -0,0 +1,23 @@
+extends Node2D
+
+var agent_scene = preload("res://scenes/agents/agent.tscn")
+var range_of_mutations = 16
+
+func _ready():
+	var agent_root_node = get_tree().root.find_child("AgentRootNode", true, false)
+	for agent_type in Cell.TYPES:
+		var num_nodes = randi() % 10 + 1  # Returns random integer between 1 and 10
+		
+		# Generate nodes
+		for i in range(num_nodes):
+			var color_code = randi() % 6  # Returns random integer between 0 and 3
```

Comment:
The comment is wrong, it returns a random number between 0 and 5. But should it not return a random number between 0 and `range_of_mutations - 1`, so `var color_code = randi() % range_of_mutations`?